src/builder/builder.js

```diff
@@ -0,0 +1,128 @@
+'use strict'
+
+const extend = require('deep-extend')
+const assert = require('assert')
+const UnixFS = require('ipfs-unixfs')
+const pull = require('pull-stream')
+const parallel = require('async/parallel')
+const waterfall = require('async/waterfall')
+const dagPB = require('ipld-dag-pb')
+const CID = require('cids')
+
+const reduce = require('./reduce')
+
+const DAGNode = dagPB.DAGNode
+
+const defaultOptions = {
+  chunkerOptions: {
+    maxChunkSize: 262144
+  }
+}
+
+module.exports = function (Chunker, ipldResolver, Reducer, _options) {
```

Comment:
Please use uppercase variables only for constructors, and if they are constructors call them with `new`.